multi/imxrt-multi/uart.c

```diff
@@ -121,6 +121,9 @@ static void uart_intrThread(void *arg)
 			condWait(uart->cond, uart->lock, 0);
 		}
 
+		/* diable RX & TX interrupts */
+		*(uart->base + ctrlr) &= ~((1 << 23) | (1 << 21));
+
```

Comment:
I don't get why you would want to disable RX irq. Where is it reenabled?

Reply:
Don't know, RX and TX are disabled in interrupt handler, after returning from IRQ handler UART thread wakes up. Somehow in handling the IRQ, if the framing error and noise bits are set (both not one), the RX interrupt does not turn off, and if a lot of data comes quickly, there is a deadlock between the IRQ and UART thread, the libtty does not empty the FIFO on time and overflow occurs, RX IRQ arrives with a framing error and noisy flags set, RX can't be turned off in the interrupt there, wake up the thread. And again, again. System dies in a loop.

This was an attempt to limit this behavior, before rewriting the baudrate calculation (I noticed that the deviation from the nominal 460k and 230k speeds I use may introduce framing errors), I left this line for discussion, some sort of solution to the problem I described. Currently the behaviour has gone with proposed baud rate calculation, but I'm using 230k/460k most of the time. Similar problems may appear in the case of opto-ports, but at lower speeds, e.g. 9600 or 19200, and deviations from the nominal transmission speed of foreign devices (readers, optical heads). I can remove this line (as I also see the typo in the comment :)